tests/Unit/NumericalAlgorithms/LinearSolver/Actions/Test_TerminateIfConverged.cpp

```diff
@@ -0,0 +1,91 @@
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "NumericalAlgorithms/LinearSolver/Actions/TerminateIfConverged.hpp"  // IWYU pragma: keep
+#include "NumericalAlgorithms/LinearSolver/Tags.hpp"  // IWYU pragma: keep
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+#include "tests/Unit/ActionTesting.hpp"
+
+namespace {
+
+using simple_tags = db::AddSimpleTags<LinearSolver::Tags::HasConverged>;
+
+template <typename Metavariables>
+struct ElementArray {
+  using metavariables = Metavariables;
+  using chare_type = ActionTesting::MockArrayChare;
+  using array_index = int;
+  using const_global_cache_tag_list = tmpl::list<>;
+  using action_list = tmpl::list<LinearSolver::Actions::TerminateIfConverged>;
+  using initial_databox = db::compute_databox_type<simple_tags>;
+};
+
+struct Metavariables {
+  using component_list = tmpl::list<ElementArray<Metavariables>>;
+  using const_global_cache_tag_list = tmpl::list<>;
+};
+
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Numerical.LinearSolver.Actions.TerminateIfConverged",
+                  "[Unit][NumericalAlgorithms][LinearSolver][Actions]") {
+  using MockRuntimeSystem = ActionTesting::MockRuntimeSystem<Metavariables>;
+  using MockDistributedObjectsTag =
+      MockRuntimeSystem::MockDistributedObjectsTag<ElementArray<Metavariables>>;
+
+  const int self_id{0};
+
+  MockRuntimeSystem::TupleOfMockDistributedObjects dist_objects{};
+
+  SECTION("ProceedIfNotConverged") {
+    tuples::get<MockDistributedObjectsTag>(dist_objects)
+        .emplace(self_id, db::create<simple_tags>(false));
+    MockRuntimeSystem runner{{}, std::move(dist_objects)};
+    const auto get_box = [&runner, &self_id]() -> decltype(auto) {
+      return runner.algorithms<ElementArray<Metavariables>>()
+          .at(self_id)
+          .get_databox<ElementArray<Metavariables>::initial_databox>();
+    };
+    {
+      const auto& box = get_box();
+      CHECK(db::get<LinearSolver::Tags::HasConverged>(box) == false);
+    }
+
+    runner.next_action<ElementArray<Metavariables>>(self_id);
+
+    const auto& box = get_box();
+    CHECK(db::get<LinearSolver::Tags::HasConverged>(box) == false);
+    CHECK(runner.algorithms<ElementArray<Metavariables>>()
+              .at(self_id)
+              .get_terminate() == false);
+  }
+  SECTION("TerminateIfConverged") {
+    tuples::get<MockDistributedObjectsTag>(dist_objects)
+        .emplace(self_id, db::create<simple_tags>(true));
+    MockRuntimeSystem runner{{}, std::move(dist_objects)};
+    const auto get_box = [&runner, &self_id]() -> decltype(auto) {
+      return runner.algorithms<ElementArray<Metavariables>>()
+          .at(self_id)
+          .get_databox<ElementArray<Metavariables>::initial_databox>();
+    };
+    {
+      const auto& box = get_box();
+      CHECK(db::get<LinearSolver::Tags::HasConverged>(box) == true);
+    }
+
+    runner.next_action<ElementArray<Metavariables>>(self_id);
+
+    const auto& box = get_box();
```

Comment:
Why not just write `CHECK(db::get<LinearSolver::Tags::HasConverged>(get_box()));`?